modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators/MustBeVariantAnalyzer.cs

```diff
@@ -111,7 +111,9 @@ private bool ShouldCheckTypeArgument(SyntaxNodeAnalysisContext context, SyntaxNo
         {
             var typeParamSymbol = parentSymbol switch
             {
-                IMethodSymbol methodSymbol => methodSymbol.TypeParameters[typeArgumentIndex],
+                IMethodSymbol methodSymbol => methodSymbol.TypeParameters.Length > 0
+                    ? methodSymbol.TypeParameters[typeArgumentIndex]
+                    : methodSymbol.ContainingType.TypeParameters[typeArgumentIndex],
```

Comment:
I feel like this should also assert that the method is an attribute constructor, otherwise it may have unintended consequences with methods of other generic classes. To do this maybe we can check if `parentSyntax` is `AttributeSyntax`.

Reply:
Done! Let me know if checking for `ExpressionSyntax` is redundant.

Reply:
Yes, I think it's a bit redundant.

Reply:
Done! :)